src/hazelcore/HTyp.re

```diff
@@ -93,21 +99,21 @@ let get_prod_arity = ty => ty |> get_prod_elements |> List.length;
 /* matched sum types */
 let matched_sum =
   fun
-  | Hole => Some((Hole, Hole))
+  | Unknown(prov) => Some((Unknown(prov), Unknown(prov)))
   | Sum(tyL, tyR) => Some((tyL, tyR))
   | _ => None;
 
 /* matched list types */
 let matched_list =
   fun
-  | Hole => Some(Hole)
+  | Unknown(prov) => Some(Unknown(prov))
   | List(ty) => Some(ty)
   | _ => None;
 
 /* complete (i.e. does not have any holes) */
```

Comment:
I don't think we use complete anywhere, and its a bit underspecified here since its detecting all unknowns not just holes. Assuming it isn't actually used anywhere lets go ahead and remove this. We need only define it for UHTyp, not HTyp.